### DISCUSSION TITLE
1.3.4

### DISCUSSION THREAD

**Discussion by betula** [betula/use-between#35] [General]
Hello everyone!

React 18 was recently released and I began to see mentions of concurrent rendering on the Internet more often. We are all very excited about the new release and want to be ready! One of the checks to make sure your application is ready for a new rendering mode is to make sure it works in Strict Mode.

In release 1.3.4 I added support for Strict Mode - now everything works great. https://github.com/betula/use-between/issues/33 

You can keep improving your projects and keep up with the times.

Enjoy your code!

<hr /><em>This discussion was created from the release <a href='https://github.com/betula/use-between/releases/tag/v1.3.4'>1.3.4</a>.</em>

**Comment by danjguzman** [comment]
This is a great idea and I love how it's implemented in the easiest way possible, which so far is the best out of all these share-state libraries.

However...

If you haven't already, check out "**Jotai**" which does something similar, albiet a little bit more complex. However it has one great advantage: When the state changes in **Jotai** atoms, it only re-renders the components using these states, and not the entire app.

With **use-between**, I noticed that when I use it in **ComponentA** and **ComponentB** only, **ComponentC** will still trigger a re-render when those shared states are changed. 

Is there any plans to eliminate unnecessary re-renders in **use-between** like in **Jotai**?

**Reply by betula** [reply]
Hello @danjguzman,

Many many thanks for your warm words! I’m so glad to hear it, and wish us a lot of years of growth in our beautiful community))

useBetween develops its performance as it grows! But what you are talking about is already implemented, only those components are updated that use the data that has changed.

Can you show your example, I tried to reproduce, but I have a different behavior. 
I made “useCounter” hook, and connect It between two React components “Count” and “Buttons”. And only they were updated for every change in the shared “useCounter".

https://codesandbox.io/s/rerender-check-usebetween-ozvm98?file=/src/App.js

Thanks!

**Comment by danjguzman** [comment]
Never mind on this issue, I realized I had some old code when I tested use-between that was causing a re-render from something else unrelated to use-between.

**Comment by danjguzman** [comment]
I've been using useBetween for over a month now and I've changed the way I develop components and hooks because of this. It's excellent, easy, and fast. And my code is much smaller. It would be great if React included this built-in as a default Hook, it's better than every State manager I have used in the past. Excellent work.

**Reply by betula** [reply]
Hi @danjguzman,
Many many thanks for your great words! Me too! I believe one time it will be into React, it will be better future for use-between!
